### ISSUE TITLE
Failed to load structure nbt file from mostructures:empty

### ISSUE THREAD

**Issue by PseudoDistant** [frqnny/mostructures#50]
I'm running a modded server (Fabric), and this comes up when certain chunks are loaded.
```
****************** Blame Report 1.17-4.0.0-fabric ******************

 Failed to load structure nbt file from : mostructures:empty
 The path represented by this is:  data/mostructures/structures/empty.nbt
 Most common cause is that there is a typo in this path to the nbt file.
 Please let the mod author know about this so they can double check to make sure the path is correct.
 A common mistake is putting the structure nbt file in the asset folder when it needs to go in data/structures folder.
  Stacktrace of where this NBT file is being called by: [java.base/java.lang.Thread.getStackTrace(Thread.java:1610), com.telepathicgrunt.blame.main.MissingNBTBlame.addMissingNbtDetails(MissingNBTBlame.java:56), net.minecraft.class_3485.handler$zhf000$addMissingnbtDetails2(class_3485.java:1035), net.minecraft.class_3485.method_15094(class_3485.java:65), net.minecraft.class_3485.method_15091(class_3485.java:53), net.minecraft.class_3470.<init>(class_3470.java:59), io.github.frqnny.mostructures.structure.VolcanicVentStructure$Piece.<init>(VolcanicVentStructure.java:66), net.minecraft.class_3195.method_28660(class_3195.java:151), net.minecraft.class_2852.method_12392(class_2852.java:415), net.minecraft.class_2852.method_12395(class_2852.java:175), net.minecraft.class_3898.method_17256(class_3898.java:512), java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768), net.minecraft.class_1255.method_18859(class_1255.java:151), net.minecraft.class_3215$class_4212.method_18859(class_3215.java:528), net.minecraft.class_1255.method_16075(class_1255.java:125), net.minecraft.class_3215$class_4212.method_16075(class_3215.java:537), net.minecraft.class_1255.method_18857(class_1255.java:134), net.minecraft.class_3215.getChunkBlocking(class_3215.java:1251), net.minecraft.class_3215.method_12121(class_3215.java:1147), net.minecraft.class_1937.getChunkLithium(class_1937.java:2096), net.minecraft.class_1937.method_8497(class_1937.java:2082), net.minecraft.class_1937.method_8320(class_1937.java:398), net.minecraft.class_1297.method_23312(class_1297.java:784), net.minecraft.class_2048.method_8909(class_2048.java:119), net.minecraft.class_215.method_914(class_215.java:40), net.minecraft.class_215.test(class_215.java:17), net.minecraft.class_2048$class_5258.method_27806(class_2048.java:438), net.minecraft.class_4558.method_22510(class_4558.java:64), net.minecraft.class_2092.method_9027(class_2092.java:35), net.minecraft.class_3222.method_14226(class_3222.java:522), net.minecraft.class_3244.method_18784(class_3244.java:212), net.minecraft.class_2535.method_10754(class_2535.java:238), net.minecraft.class_3242.method_14357(class_3242.java:183), net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:902), net.minecraft.class_3176.method_3813(class_3176.java:335), net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:831), net.minecraft.server.MinecraftServer.handler$bgg000$modifiedRunLoop(MinecraftServer.java:5890), net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:676), net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:270), java.base/java.lang.Thread.run(Thread.java:833)]
```
I can provide a mod list if you believe this is a conflict, and as always I'm willing to accept that I am simply stupid and just misconfigured something.

**Comment by frqnny** [comment]
this is not a bug pls stop reporting this fr fr

**Comment by RDKRACZ** [comment]
![Screenshot from 2021-07-31 17-28-48](https://user-images.githubusercontent.com/43277609/127744736-a8ac2575-129a-4a38-b04e-1044fdf89425.png)
...

**Comment by frqnny** [comment]
i have my rights!

**Comment by K0-RR** [comment]
I still have the screenshot but I will stop first. No need to delete any conversations/evidence. Why be angry just because someone reported something which looks like a bug?

**Comment by PseudoDistant** [comment]
> this is not a bug pls stop reporting this fr fr

Was it reported before?

**Comment by frqnny** [comment]
this is the sixth time this was reported. im fed up with these reports, like, stop looking at your log nothing cool in there anyway???

**Comment by frqnny** [comment]
pls check old bug reports??

**Comment by PseudoDistant** [comment]
> this is the sixth time this was reported. im fed up with these reports, like, stop looking at your log nothing cool in there anyway???

The server crashed...
And crashed again when the same player who crashed it joined...
I think it's worth looking into...

**Comment by frqnny** [comment]
nope its not a crash

**Comment by PseudoDistant** [comment]
The server refused to connect players, and stopped running a few seconds after it printed that... Also it created a file in the crash-reports folder...

**Comment by PseudoDistant** [comment]
[crash-2021-07-10_15.10.27-client.txt](https://github.com/frqnny/mostructures/files/6911414/crash-2021-07-10_15.10.27-client.txt)

**Comment by frqnny** [comment]
this has NOTHING to do with any crash. stop using blame.

**Comment by PseudoDistant** [comment]
Did Blame cause the second crash? O_O
(Honestly I don't know, I'm trying to debug.)

**Comment by frqnny** [comment]
report to viafabric-mc117 looks like crash says its theirs

**Comment by PseudoDistant** [comment]
Also again, I'm willing to accept that I simply misconfigured something, if you have any suggestions I'm willing to make the effort.
Also I'm not using ViaFabric.
It's disabled, I don't know why it comes up.

**Comment by PseudoDistant** [comment]
Wait no I dumbed, that's the wrong file (client.)
[crash-2021-07-31_10.25.37-server.txt](https://github.com/frqnny/mostructures/files/6911419/crash-2021-07-31_10.25.37-server.txt)

**Comment by frqnny** [comment]
Hmmm, there’s nothing useful in that crash. Show latest.log

**Comment by PseudoDistant** [comment]
[2021-07-31-6.log](https://github.com/frqnny/mostructures/files/6911430/2021-07-31-6.log)
(I removed the mod to see if it would crash without it, so latest.log isn't helpful.)

**Comment by frqnny** [comment]
Hmmm, remove Blame?

**Comment by PseudoDistant** [comment]
> Hmmm, remove Blame?

It hangs, and just says this on the bottom:
```
[10:25:21] [User Authenticator #4/INFO]: UUID of player FarlanderCraft is 7a8ac3c3-ce12-4cd6-adb9-67e6bcf80b64
[10:25:33] [User Authenticator #5/INFO]: UUID of player LemonadeExtract is b051b1b5-d53e-4774-b52f-285778732e12
[10:25:37] [Server Watchdog/FATAL]: A single server tick took 60.00 seconds (should be max 0.05)
[10:25:37] [Server Watchdog/FATAL]: Considering it to be crashed, server will forcibly shutdown.
[10:25:37] [Server Watchdog/ERROR]: This crash report has been saved to: /home/farlandercraft/Vanilla+ server test/./crash-reports/crash-2021-07-31_10.25.37-server.txt
```

**Comment by PseudoDistant** [comment]
Removing Mo' Structures and reloading the world led to it working properly, but without Mo' Structures. ;-;

**Comment by frqnny** [comment]
its not mo structures causing it, ill tell you that.

**Comment by frqnny** [comment]
i cant figure it out

**Comment by PseudoDistant** [comment]
I'm not saying it's explicitly a problem with your mod, I was hoping that it was a known conflict or something. The mod seems to work fine on its own in Single Player.

**Comment by frqnny** [comment]
it also works fine on aof 4

**Comment by PseudoDistant** [comment]
The only reason why I made the issue here was there wasn't much of any other information in the crash report that I could find, so I couldn't see anything from any other mods.

**Comment by frqnny** [comment]
i cant figure it out, ask in discord for better support

**Comment by frqnny** [comment]
![A378B20B-3C9B-4849-BCB6-A7917DCB0E97](https://user-images.githubusercontent.com/45723631/127745684-b2a3dd6d-84b1-4231-bfb9-d37a557206c8.jpeg)

**Comment by TelepathicGrunt** [comment]
@FarLanderCraft can you give me the latest.log file with blame on instead after you get the crash? crashlog often seem to skip out on important info which is why latest.log files are much better to share. 

However, `A single server tick took 60.00 seconds (should be max 0.05)` sounds like something deadlocked or stalled your server which causes the watchdog to think the server is stuck and thus, crash it to get it unstuck. You can try getting a thread dump when you noticed the server is hanging by following this video https://streamable.com/n4hw6w

If you cannot run VisualVM, can you share your exported mod profile or mod folder so I can quickly put them on and do the thread dump myself?

----

And just fyi, the Blame report about the ` Failed to load structure nbt file`  is most likely caused by this line. It is not going to cause a crash. In this cause, not spawning any nbt file is intentional by frqnny. Though there are ways to rewrite it slightly so it doesn't need the non-existent nbt file. Or maybe just save a zero sized nbt file at the mostructures:empty spot so mc can find it and do nothing like intended and without triggering blame 
https://github.com/frqnny/mostructures/blob/c6f5375acaec72fae26822a084858e9f98e439d4/src/main/java/io/github/frqnny/mostructures/structure/VolcanicVentStructure.java#L54

**Comment by PseudoDistant** [comment]
@TelepathicGrunt 
After a lot of trying, I found the problem. It is in fact SimplyImprovedTerrain.
I was able to reproduce the crash four times out of 25, and in each world where it occurred it reoccurred when attempting to reload the world. However without SimplyImprovedTerrain it wasn't reproducable in 50 attempts.
After all of the issues I've been having with it I would advise against using SimplyImprovedTerrain with any structure mods, at least for the time being.
